esbuild.js

```diff
@@ -1,14 +1,37 @@
-import { build } from 'esbuild'
+import { build, context } from 'esbuild'
 
-build({
-  entryPoints: ['./src/index.js'],
+const ctx = context({
+  entryPoints: ['./src/index.ts'],
   outfile: './dist/index.js',
   sourcemap: 'both',
   minify: false,
   bundle: true,
+  platform: 'browser',
+  target: ['es2020'],
+  loader: { '.ts': 'ts' },
   define: {
     'process.env.NODE_DEBUG': 'false',
     global: 'globalThis'
-  }
+  },
 })
-  .catch(() => process.exit(1))
+
+// Check command line arguments
+const isWatch = process.argv.includes('--watch')
+const isServe = process.argv.includes('--serve')
+
+if (isWatch || isServe) {
+  // Create a context for either watch or serve mode
+  ctx.then(async (ctx) => {
+    // Start serve mode if requested
+    if (isServe) {
+      const { host, port } = await ctx.serve({ servedir: './dist', })
+      console.log(`Server running at http://${host}:${port}`)
+    } else {
+      await ctx.watch()
+      console.log('Watching for changes...')
+    }
+  })
+} else {
+  // Regular one-time build
+  ctx.then(ctx => ctx.build()).catch(() => process.exit(1))
```

Comment:
without this, a misconfigured esbuild won't output anything
```suggestion
  ctx.then(ctx => ctx.build()).catch((err) => {
	console.error(err)
  	process.exit(1)
  })
```